src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -147,15 +147,34 @@ private static bool IsNamespaceName(NameSyntax name)
             }
 
             // Use .Equals since we can't rely on object identity for constructed types.
-            if (symbol is ITypeSymbol typeSymbol)
+            SyntaxToken token;
+            switch (symbol)
             {
-                var classification = GetClassificationForType(typeSymbol);
-                if (classification != null)
-                {
-                    var token = name.GetNameToken();
-                    classifiedSpan = new ClassifiedSpan(token.Span, classification);
+                case ITypeSymbol typeSymbol:
+                    var classification = GetClassificationForType(typeSymbol);
+                    if (classification != null)
+                    {
+                        token = name.GetNameToken();
+                        classifiedSpan = new ClassifiedSpan(token.Span, classification);
+                        return true;
+                    }
+                    break;
+                case IFieldSymbol fieldSymbol:
```

Comment:
Should we do local/parameter as well, since we're here?

Reply:
Should const/non-const/enum fields get their own classifications?  Probably yes to me.

Reply:
Added locals, parameters, const, enum fields.